### DISCUSSION TITLE
How to theme using CSS variables?

### DISCUSSION THREAD

**Discussion by DidacGit** [tailwindlabs/tailwindcss#10745] [Help]
I have my website's theme configured using CSS variables. For example, for colors I have now configured it with tailwind like:

```
module.exports = {
    theme: {
        colors: {
            prim: 'rgb(var(--col-prim) / <alpha-value>)',
            sec: 'rgb(var(--col-sec) / <alpha-value>)',
```
And works perfectly, I can change those values dynamically and my front-end changes.

How can I do this with other kind of themeing? For example, for cards to have rounded corners or straight ones. I would like to have something like this:

```
<div class="var(--card-classes)">
  Some card
</div>
```

And just have some utility classes in that variable: e.g. `"border rounded"`, then, just like with the colors, I could change it dynamically and would work perfectly. But I can't expand variables in the `class` property, can I?

Do you guys know how should I approach this?
Thanks a lot in advance!

**Comment by wongjn** [comment]
You would leverage your templating engine, for example, in JavaScript:
```js
let cardClasses = 'border rounded';

if (foo) {
  cardClasses += ' border-red-500';
}

return `
<div class="${cardClasses}">
  Some card
</div>
`

**Reply by DidacGit** [reply]
Thanks a lot for responding!
The thing is that I don't have a templating engine. I write my components directly with HTML.

**Reply by wongjn** [reply]
There is no HTML-native way to dynamically assign classes in a dynamic manner akin to CSS values with CSS variables. You'd have to have to use some kind of CSS variable system, like:
```css
.card {
  --border-width: 0;
  --border-radius: 0;
  border-width: var(--border);
  border-radius: var(--border-radius);
}
```
```html
<div class="card [--border-width:theme(borderWidth.DEFAULT)] [--border-radius:theme(borderRadius.DEFAULT)]">
  Some card
</div>
<div class="card" style="--border-width:5px">
  Some card
</div>
<div class="card" style="--border-radius:10px">
  Some card
</div>
```
